Mage.Sets/src/mage/cards/v/VrestinMenoptraLeader.java

```diff
@@ -0,0 +1,74 @@
+package mage.cards.v;
+
+import mage.MageInt;
+import mage.abilities.common.AttacksWithCreaturesTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.dynamicvalue.common.GetXValue;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.EntersBattlefieldWithXCountersEffect;
+import mage.abilities.effects.common.counter.AddCountersAllEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.counters.CounterType;
+import mage.filter.common.FilterAttackingCreature;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.permanent.token.AlienInsectToken;
+
+import java.util.UUID;
+
+/**
+ * @author Sidorovich77
+ */
+public final class VrestinMenoptraLeader extends CardImpl {
+
+    private static final FilterAttackingCreature filter = new FilterAttackingCreature();
```

Comment:
fyi this can just be a FilterPermanent (the attacking and creature aspects are handled by the trigger)